### PR TITLE
Add a test to ensure addons can't be named Ember or ember-cli

### PR DESCRIPTION
**Pull request by johnotander**
Related #5487

### CODE REVIEW

File: tests/unit/commands/addon-test.js [hunk 1]
```diff
@@ -30,13 +30,31 @@ describe('addon command', function() {
...
+      expect(false, 'should have rejected with an addon name of ember');
     })
     .catch(function(error) {
       expect(error.message).to.equal('We currently do not support a name of `ember`.');
     });
   });
 
+  it('doesn\'t allow to create an addon named `Ember`', function() {
+    return command.validateAndRun(['Ember']).then(function() {
+      expect(false, 'should have rejected with an addon name of Ember');
```
Comment (Turbo87) [inline tests/unit/commands/addon-test.js]: `expect(false)` is not actually asserting anything, you need to use something like `.to.be.true` at the end. see #5483

Comment (johnotander) [inline tests/unit/commands/addon-test.js]: Ah, thanks, that makes sense. I've gone ahead and fixed those.

After a quick grep I noticed that this `expect(false, 'some string');` pattern is repeated throughout the tests. So, I can go ahead and update those, too. I'm assuming I should make that a separate PR?

Comment (Turbo87) [inline tests/unit/commands/addon-test.js]: @johnotander I've updated those in #5483 already, but thanks for the offer!

Comment (johnotander) [inline tests/unit/commands/addon-test.js]: Whoops, missed that part :blush: . Awesome!

**Comment by Turbo87** [comment]
r+ from my side. @kellyselden r?

**Comment by kellyselden** [comment]
@Turbo87 other than you needing a rebase on those other test changes, LGTM.

**Comment by kellyselden** [comment]
@homu r+

**Comment by kellyselden** [comment]
Thank you @johnotander

**Pull request merged**